src/platform/datapath_raw_xdp.c

```diff
@@ -66,26 +67,41 @@ typedef struct XDP_INTERFACE {
     uint8_t RuleCount;
     CXPLAT_LOCK RuleLock;
     XDP_RULE* Rules;
-    XDP_QUEUE* Queues;
+    XDP_QUEUE* Queues; // An array of queues.
 } XDP_INTERFACE;
 
-typedef struct XDP_DATAPATH {
-    CXPLAT_DATAPATH;
-
-    BOOLEAN Running;
+typedef struct QUIC_CACHEALIGN XDP_WORKER {
+    const struct XDP_DATAPATH* Xdp;
     HANDLE CompletionEvent;
+    XDP_QUEUE* Queues; // A linked list of queues, accessed by Next.
+    uint16_t ProcIndex;
+} XDP_WORKER;
+
+void XdpWorkerAddQueue(_In_ XDP_WORKER* Worker, _In_ XDP_QUEUE* Queue) {
+    XDP_QUEUE** Tail = &Worker->Queues;
+    while (*Tail != NULL) {
+        Tail = &(*Tail)->Next;
+    }
+    *Tail = Queue;
```

Comment:
Is there some particular reason the new queue is added to the tail instead of the head?

Reply:
Not really, I guess. I just tried to keep them in order that I created them, but I don't have any good justification requiring that.